recap/src/Imp/Eval.hs

```diff
@@ -17,14 +20,37 @@ data Error = UndefinedVar String | ParsingErr String | DivByZero deriving (Show)
 -- Evaluation monad
 type EvalM = StateT VarMap (ExceptT Error IO)
 
+mapOp :: M.Map Op (Int -> Int -> Int)
+mapOp = M.fromList [(Plus, (+)), (Minus, (-)), (Mult, (*)), (Div, div)]
+
 evalExpr :: Expr -> EvalM Int
-evalExpr = undefined 
+evalExpr (Var x) = do
+  mapVar <- get
+  when (M.notMember x mapVar) $ lift (throwE $ UndefinedVar x)
+  return $ mapVar M.! x
```

Comment:
Here, you search in the map twice. It's not critical, but can be avoided by using `lookup` and pattern matching (or using `maybe`.